File: 53-MaximumSubarray/53-MaximumSubarray.java
// Last updated: 03/06/2025, 09:54:59
class Solution {
    public int maxSubArray(int[] arr) {
        int mehere = arr[0];  
        int msfar = arr[0]; 

        for (int i = 1; i < arr.length; i++) {
            mehere = Math.max(arr[i], mehere + arr[i]);
            msfar = Math.max(msfar, mehere);
        }

        return msfar;
    }
}

File: 53-MaximumSubarray/53-MaximumSubarray.cpp
// Last updated: 03/06/2025, 09:55:06
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = INT_MIN , currSum = 0 ;
        for(int val : nums){
            currSum = currSum + val;
            maxSum = max( currSum , maxSum);
            if ( currSum < 0)
            currSum = 0;
        } 
        return maxSum ;
    }
};